invenio_rdm_records/fixtures/vocabularies.py

```diff
@@ -106,84 +109,249 @@ def __iter__(self):
+            filepath = directory / self._filename
+            self.load_vocabularies(filepath)
+
+    def peek_vocabularies(self, filepath):
+        """Peek at vocabularies listed in vocabularies file.
+
+        Returns list of vocabularies.
+        """
+        vocabularies = []
+        fixture = VocabulariesFixture(self._identity, filepath)
+        for id_, entry in fixture.read():
+            if isinstance(entry["data-file"], dict):
+                subvocabularies = [
+                    f"{id_}.{k}" for k in entry["data-file"].keys()
+                ]
+                vocabularies.extend(subvocabularies)
+            else:
+                vocabularies.append(id_)
+        return vocabularies
+
+    def load_vocabularies(self, filepath):
+        """Load vocabularies listed in vocabularies file."""
+        fixture = VocabulariesFixture(
+            self._identity,
+            filepath,
+            delay=self._delay
+        )
+        self._loaded_vocabularies = fixture.load(
+            ignore=self._loaded_vocabularies
+        )
+
+
+class VocabulariesFixture:
+    """Vocabularies fixture.
+
+    This class' responsibility is to load the vocabularies in its
+    vocabularies file.
+    """
+
+    def __init__(self, identity, filepath, delay=True):
+        """Initialize the fixture."""
+        self._filepath = filepath
+        self._identity = identity
+        self._delay = delay
+
+    def read(self):
+        """Return content of vocabulary file."""
+        dir_ = self._filepath.parent
+        with open(self._filepath) as f:
+            data = yaml.safe_load(f) or {}
+            for id_, entry in data.items():
+                if isinstance(entry["data-file"], dict):
+                    entry["data-file"] = {
+                        k: dir_ / v for k, v in entry["data-file"].items()
+                    }
+                else:
+                    entry["data-file"] = dir_ / entry["data-file"]
+
+                yield id_, entry
 
     def get_records_by_vocabulary(self, vocabulary_id):
         """Get all records of a given vocabulary."""
-        for id_, entry in self._open_vocabularies():
+        for id_, entry in self.read():
             if vocabulary_id != id_:
                 continue
             for record in self.iter_datafile(entry["data-file"]):
                 yield record
 
-    def load(self):
+    def load(self, ignore=None):
         """Load the fixture.
 
-        Fixtures found sooner in self._search_paths have priority.
+        ignore: iterable of ids to ignore
+
+        For subjects (or any vocabulary with a dict of data-file), the id
+        that counts is ``<vocabulary id>.<dict key>``.
         """
-        ids = set()
+        ids = set().union(ignore) if ignore else set()
 
-        for id_, entry in self._open_vocabularies():
+        for id_, entry in self.read():
+            # TODO: We need to carefully think about what the workflow for
```

Comment:
What would be an action for this TODO? e.g. is not clear to me what's expected of someone finding this... just wondering if it can be removed and is a bigger taks i.e. issue.

I think tend to forget TODO/FIXME in general...

Reply:
Yeah... I am keeping track of those in my personal notes but I should really create issues for them. I created the following:

- Discussion about other use cases around subject loading: https://github.com/inveniosoftware/invenio-rdm-records/issues/687 . I will bring this up in the chat
- A concrete issue about using the content of database to enforce priority (needs other PR in invenio-vocabularies first): https://github.com/inveniosoftware/invenio-rdm-records/issues/686